lib/modules/blockchain_process/blockchain.js

```diff
@@ -178,9 +179,12 @@ Blockchain.prototype.run = function() {
       console.error(`Geth error: ${data}`);
     });
     // Geth logs appear in stderr somehow
-    self.child.stderr.on('data', (data) => {
+    self.child.stderr.on('data', async (data) => {
```

Comment:
Should we move this to a `.once('data', ...)` (as opposed to `.on('data'), ...`, so that the proxy setup only fires once?

Reply:
I inferred that `data.indexOf('WebSocket endpoint opened')` would only be true at most one time, maybe that's incorrect. Thoughts?

Reply:
Fair point, does the proxy only work for websockets?

Reply:
Good question!  The conditional pre-dates this PR, but you've made me curious whether it would have an effect if only http rpc is enabled (ws is disabled).

Reply:
Geth output after start looks like the following:

> Geth: INFO [09-20|13:48:45.108] IPC endpoint opened                      url=/Users/emizzle/Code/__Github/embk-fw/embark/templates/demo/.embark/development/datadir/geth.ipc
Geth: INFO [09-20|13:48:45.128] HTTP endpoint opened                     url=http://localhost:8555                                                                           cors=http://localhost:8000,http://localhost:8080,embark vhosts=localhost
Blockchain node is ready
Geth: INFO [09-20|13:48:45.133] WebSocket endpoint opened                url=ws://127.0.0.1:8556

As long as geth is configured to always allows for websockets, and under the assumption that geth initialises WebSockets last, then this should be sufficient. What happens if we switch it up and don't start websockets, or geth output order changes? In my opinion, I think a more complete way to handle this would be to parse the output, and start the corresponding proxy if geth is listening, ie

``` 
if (data.indexOf('WebSocket endpoint opened') > -1){
  self.setupProxy('ws');
}
if (data.indexOf('HTTP endpoint opened') > -1){
  self.setupProxy('http');
}
```

Reply:
In reading this as  well, it appears that "blockchain node is ready" happen after the HTTP endpoint is opened up, and not websockets, I wonder if that has any side effects 🤔

Reply:
Timing snafu, for sure. Good work sniffing it out. Feel free to push commits to this branch/PR if you hit on a solution. I'll check back (my) tomorrow.

Reply:
Updated (below).